rust/main/agents/validator/src/submit.rs

```diff
@@ -127,6 +138,61 @@ impl ValidatorSubmitter {
             })
             .await;
 
+            if latest_checkpoint.index < latest_seen_checkpoint_index {
+                if let Some(block_height) = latest_checkpoint.block_height {
+                    if block_height >= latest_seen_checkpoint_block_height {
```

Comment:
Since we can detect reorg only if we have block height available, we can rearrange this ifs so that we can combine two of them into single if condition, but making `if let` the most outer one.